RubeesEat/Model/DB/DbBillRepository.cs

```diff
@@ -0,0 +1,277 @@
+        command.CommandType = CommandType.Text;
+
+        connection.Open();
+
+        using var reader = command.ExecuteReader();
+
+        var bills = new List<Bill>();
+        while (reader.Read())
+        {
+            var guid = reader.GetGuid(0);
+            bills.Add(new Bill
+            (
+                guid,
+                reader.GetDateTime(1),
+                reader.GetString(2),
+                [..GetEntryLines(guid)]
+            ));
+        }
+
+        return bills.AsReadOnly();
+    }
+
+    private EntryLine[] GetEntryLines(Guid billID)
+    {
+        var entryLines = new List<EntryLine>();
+
+        using var connection = connectionFactory.CreateDbConnection();
+        using var command = connection.CreateCommand();
+
+        command.CommandText = "SELECT PersonID, Amount FROM EntryLines WHERE BillID = @BillID;";
+        command.CommandType = CommandType.Text;
+
+        var billIdParam = command.CreateParameter();
+        billIdParam.ParameterName = "@BillID";
+        billIdParam.Value = billID;
+        command.Parameters.Add(billIdParam);
+
+        connection.Open();
+
+        using var reader = command.ExecuteReader();
+
+
+        while (reader.Read())
+        {
+            var guid = reader.GetGuid(0);
+            entryLines.Add(new EntryLine(
+                GetPersonById(guid),
+                reader.GetDecimal(1)
+            ));
+        }
+
+        return entryLines.ToArray();
+
+
+        Person GetPersonById(Guid guid)
+        {
+            var personRepository = new DbPersonRepository(connectionFactory);
+            return personRepository.GetById(guid);
+        }
+    }
+
+    public void Add(Bill bill)
+    {
+        using (var connection = connectionFactory.CreateDbConnection())
+        {
+            using (var command = connection.CreateCommand())
+            {
+                command.CommandText = "INSERT INTO Bills (BillID, Date, Description) VALUES (@BillID, @Date, @Description);";
+                command.CommandType = CommandType.Text;
+
+                var billIdParam = command.CreateParameter();
+                billIdParam.ParameterName = "@BillID";
+                billIdParam.Value = bill.Id;
+                command.Parameters.Add(billIdParam);
+
+                var dateParam = command.CreateParameter();
+                dateParam.ParameterName = "@Date";
+                dateParam.Value = bill.Date;
+                command.Parameters.Add(dateParam);
+
+                var descriptionParam = command.CreateParameter();
+                descriptionParam.ParameterName = "@Description";
+                descriptionParam.Value = bill.Description;
+                command.Parameters.Add(descriptionParam);
+
+                connection.Open();
+                command.ExecuteNonQuery();
+            }
+        }
+
+        foreach (var entryLine in bill.EntryLines)
+        {
+            AddEntryLine(bill.Id, entryLine);
+        }
```

Comment:
The inserted `Bill` and `EntryLine` instances need to be inserted as part of a transaction. Otherwise, it might be possible to read a `Bill` which does not have any `EntryLine`s or that the `EntryLine`s do not add to 0. Please use transactions here and choose a fitting isolation mode for the transaction. For the transaction to work we need to do all the work in the same connection.